Spec/RestClient.swift

```diff
@@ -1384,71 +1384,35 @@ class RestClient: QuickSpec {
                 }
             }
 
-        }
-        
-        context("if in the course of a REST request an attempt to authenticate using authUrl fails due to a timeout") {
-            // RSA4e
-            it("the request should result in an error with code 40170, statusCode 401, and a suitable error message") {
-                let options = AblyTests.commonAppSetup()
-                let token = getTestToken()
-                options.httpRequestTimeout = 3 // short timeout to make it fail faster
-                options.authUrl = NSURL(string: "http://10.255.255.1")! as URL
-                options.authParams = [NSURLQueryItem]() as [URLQueryItem]?
-                options.authParams?.append(NSURLQueryItem(name: "type", value: "text") as URLQueryItem)
-                options.authParams?.append(NSURLQueryItem(name: "body", value: token) as URLQueryItem)
-                
-                let client = ARTRest(options: options)
-                waitUntil(timeout: testTimeout, action: { done in
-                    let channel = client.channels.get("test-channel")
-                    channel.publish("test", data: "test-data", callback: { error in
-                        guard let error = error else {
-                            fail("Error should not be empty")
+            context("if in the course of a REST request an attempt to authenticate using authUrl fails due to a timeout") {
+                // RSA4e
+                it("the request should result in an error with code 40170, statusCode 401, and a suitable error message") {
+                    let options = AblyTests.commonAppSetup()
+                    let token = getTestToken()
+                    options.httpRequestTimeout = 3 // short timeout to make it fail faster
+                    options.authUrl = NSURL(string: "http://10.255.255.1")! as URL
```

Comment:
You can use directly the `URL` object, so `URL(string: "http://10.255.255.1")!` instead of we you have now.

Reply:
Did you moved the code to another place, is that it?

Reply:
@ricardopereira No, as part of a rebase this PR includes one of your commits https://github.com/ably/ably-ios/pull/791/commits/09c428bd4c95afde03556e4caa8641f6d1b2780a

Reply:
@ricardopereira 
> You can use directly the URL object, so URL(string: "http://10.255.255.1")! instead of we you have now.

There's many occurrences of `NSURL(...)` in the swift code. I propose to address this in a separate PR. [Issue here](https://github.com/ably/ably-ios/issues/794). Ok?